.github/workflows/release.yaml

```diff
@@ -5,37 +5,33 @@ on:
       - 'v*.*.*'
 jobs:
   build-and-push-image:
-    env:
-      IMAGE_NAME: autoinstrumentation-go
     runs-on: ubuntu-latest
     permissions:
       contents: read
       packages: write
+
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3.3.0
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
-      - name: Log in to Docker Hub
-        uses: docker/login-action@v2
-        with:
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
+      - uses: actions/checkout@v3
+
       - name: Log in to the Container registry
         uses: docker/login-action@v2
         with:
           registry: ghcr.io
-          username: ${{ github.actor }}
+          username: ${{ github.repository_owner }}
```

Comment:
Using repository owner is the pattern the demo and operator use.  Since we're having issues following that doc for our repository (but not my personal) I am opting to follow the pattern that we know is working for other OTel repos.

Reply:
Examples:
- https://github.com/open-telemetry/opentelemetry-demo/blob/5ef82f3dbd7eca51cd439bc0fd6b14be7abde3bc/.github/workflows/release.yml#L91
- https://github.com/open-telemetry/opentelemetry-operator/blob/923f9a208afddd51f66c2d54e8c852389bacad0d/.github/workflows/publish-images.yaml#L65

I am putting a lot of stock on the operator build at the moment as it is the only other OTel repo (that I could find) that uses the metadata action that we need to use.

Reply:
My suggestion works fine in OTel .NET auto-instrumentation: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/02371f51ae8a7acde61cb01a647cb7a691286214/.github/workflows/build_and_push_image.yml#L25

Anyway I do not intend to block the PR 😉

Reply:
Here is the image created by the workflow about 3 months ago: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pkgs/container/opentelemetry-dotnet-instrumentation-centos7-build-image

This was the GitHub workflow in the time the image was created: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/36886c41781c76a38ed981827f694628a56a5f25/.github/workflows/build_and_push_image.yml

Reply:
Do you know why the action doesn't show up in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/actions/workflows/build_and_push_image.yml?

Reply:
I tried to understand it as well and I have no idea. 

1. The default retention policy is 90 days. Maybe OTel organization has it shorter? 
2. I put the wrong commit. It was https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/commit/9b647d0f6be2329e88a2070dc2badbf93f02037f. So the hyperlink should be: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/actions/workflows/push_centos_build_image.yml. Maybe the history is removed after a rename?

Reply:
Oh I bet it was the rename, good find.